src/twisted/conch/test/test_scripts.py

```diff
@@ -30,7 +30,10 @@
 else:
     try:
         tkinter.Tk().destroy()
-    except tkinter.TclError as e:
+    except (tkinter.TclError, RuntimeError) as e:
+        # On GitHub Action the macOS Python might not support the version of TK
+        # provided by the OS and it will raise a RuntimeError
+        # See: https://github.com/actions/setup-python/issues/649
```

Comment:
It seems that setup-python is prone to releasing broken tkinter releases I think it would be better to reraise the exception if not on known bad sys.version/sys.platform

Reply:
The change was done along the lines of "should Twisted continue to ship Tkinter-based UIs?"

I think that any extra effort should be done for deprecating tkinter support in Twisted.

Reply:
There are definitely users of twisted with Tkinter applications. It mostly juts works for them.

Reply:
I think the point adi meant to raise here is that supporting the Tk application that ships with *conch* may be worth re-considering at this point - not that Twisted as a whole should desupport Tk and make it impossible to write applications that use Tk and Twisted.

Reply:
Sorry for the misunderstanding.

yes. The idea is to deprecate the tk support in `twisted.conch`

We can continue to have the tk reactor.

Reply:
Oh. Yeah… difficult to make the case that tkconch is holding up any critical infrastructure these days :)

I'd love to ship a tk demo that does something interesting with Twisted but the only thing I can get tkconch to do right now is to spit out this traceback, crash, and exit when I hit 'connect':

```
Traceback (most recent call last):
  File "/Users/glyph/Projects/Twisted/src/twisted/python/log.py", line 96, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/Users/glyph/Projects/Twisted/src/twisted/python/log.py", line 80, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/Users/glyph/Projects/Twisted/src/twisted/python/context.py", line 117, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/glyph/Projects/Twisted/src/twisted/python/context.py", line 82, in callWithContext
    return func(*args, **kw)
--- <exception caught here> ---
  File "/Users/glyph/Projects/Twisted/src/twisted/internet/selectreactor.py", line 148, in _doReadOrWrite
    why = getattr(selectable, method)()
  File "/Users/glyph/Projects/Twisted/src/twisted/internet/tcp.py", line 248, in doRead
    return self._dataReceived(data)
  File "/Users/glyph/Projects/Twisted/src/twisted/internet/tcp.py", line 253, in _dataReceived
    rval = self.protocol.dataReceived(data)
  File "/Users/glyph/Projects/Twisted/src/twisted/conch/ssh/transport.py", line 764, in dataReceived
    self.dispatchMessage(messageNum, packet[1:])
  File "/Users/glyph/Projects/Twisted/src/twisted/conch/ssh/transport.py", line 781, in dispatchMessage
    f(payload)
  File "/Users/glyph/Projects/Twisted/src/twisted/conch/ssh/transport.py", line 1968, in ssh_KEX_DH_GEX_GROUP
    return self._ssh_KEX_ECDH_REPLY(packet)
  File "/Users/glyph/Projects/Twisted/src/twisted/conch/ssh/transport.py", line 1903, in _ssh_KEX_ECDH_REPLY
    d = self.verifyHostKey(hostKey, fingerprint)
  File "/Users/glyph/Projects/Twisted/src/twisted/conch/scripts/tkconch.py", line 471, in verifyHostKey
    {b"ssh-dss": "DSA", b"ssh-rsa": "RSA"}[keyType],
builtins.KeyError: b'ecdsa-sha2-nistp256'
```

Not exactly the best demonstration of our capabilities :-|